### DISCUSSION TITLE
Brew formula for Opensearch 1.x?

### DISCUSSION THREAD

**Discussion by aadmathijssen** [Homebrew/discussions#3950] [Writing Formulae/Casks] [answered]
On May 27, 2022, OpenSearch was upgraded from 1.3.2 to 2.0.0 (see [PR #102380](https://github.com/Homebrew/homebrew-core/pull/102380)). However, the 2.x release line contains breaking changes (I encountered this when I configured Magento to use OpenSearch which didn't work with version 2.x) while the 1.x release is still supported until December 31, 2023 (see the [OpenSearch maintenance policy](https://opensearch.org/releases.html#maintenance-policy)).

I think it will be a great addition if a new `opensearch@1` formula is added that can be used to install the latest 1.x version of OpenSearch.

Note that I did manage to install version 1.3.2 (see below), but I'm not sure how long this will continue to function. Also, in the meantime OpenSearch 1.3.6 has been released, and I'm unable to upgrade to that version because no brew formula exists for that version.

Steps I used to downgrade OpenSearch from 2.x to 1.3.2:

```
brew services stop opensearch
brew uninstall opensearch
rm -r $HOMEBREW_PREFIX/etc/opensearch $HOMEBREW_PREFIX/var/lib/opensearch
brew tap-new $USER/opensearch
brew extract --version=1.3.2 opensearch $USER/opensearch
brew install opensearch@1.3.2
brew services start opensearch@1.3.2
```

**Comment by aadmathijssen** [comment, marked as answer]
I have created the [isaaceindhoven/opensearch-maintenance tap](isaaceindhoven/opensearch-maintenance). With this you can install the latest version of OpenSearch 1 (1.3.6 at the time of writing) as follows:

    brew tap isaaceindhoven/opensearch-maintenance
    brew install opensearch@1

**Comment by JMLucas96** [comment]
Hi I receive the following error, can you help me please? 

`Already downloaded: .../Library/Caches/Homebrew/downloads/a4be120bf53ec8523fd3634eb727a69099494743a45e868e777e36916f97f40b--openjdk@11--11.0.28.arm64_sequoia.bottle.tar.gz
Error: gradle@6 has been disabled because it is not maintained upstream! It was disabled on 2024-12-14.
==> No outdated dependents to upgrade!`

**Reply by aadmathijssen** [reply]
Hi @JMLucas96, another user also encountered this issue, see [issue #8 of the homebrew-opensearch-maintenance repository](https://github.com/iodigital-com/homebrew-opensearch-maintenance/issues/8). Does the suggestion to force reinstall the package work?

When I try this on my dev environment, it still shows the "gradle@6 has been disabled" message, but as a warning, not as an error.

**Comment by aadmathijssen** [comment]
I have marked the `opensearch@1` formula in the [`iodigital-com/opensearch-maintenance` tap](https://github.com/iodigital-com/homebrew-opensearch-maintenance) as deprecated, as OpenSearch 1 has been marked as deprecated as of 06-05-2025; see the [OpenSearch maintenance policy](https://opensearch.org/releases/#maintenance-policy) for details.

Therefore, I am closing this discussion.

**Comment by aadmathijssen** [comment]
See discussion #6331 for OpenSearch 2.
